### DISCUSSION TITLE
Color Sample

### DISCUSSION THREAD

**Discussion by wartburggraaf** [nl-design-system/backlog#324] [Component suggestions]
<img width="480" alt="visuele weergave van een Color sample component" src="https://github.com/user-attachments/assets/1602202e-a6c3-4a74-84d7-2d30c29cc07f">

We willen graag meer leren over de Color Sample component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook, Github of screenshots. Met deze informatie kunnen we bewijzen dat de component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1725443293952589)

## Naam

Color Sample

## Doel

Toont een voorbeeld van een kleur.

## Ook bekend als

Color Swatch

## Richtlijnen

Acceptatiecriteria: [NL Design System Acceptatiecriteria voor Color Sample](https://nldesignsystem.nl/color-sample)

**Comment by jeffreylauwers** [comment]
## Amsterdam

Niet gevonden als component. Maar er worden wel voorbeelden van kleuren getoond voor het design system.

- [Storybook](https://designsystem.amsterdam/?path=/docs/brand-design-tokens-colour--docs)
- [Figma](https://www.figma.com/design/9IGm6IdPUYizBNGsUnueBd/Amsterdam-Design-System?node-id=0-1)

---

### Screenshots / Varianten

**Storybook**

![Scherm­afbeelding 2024-09-06 om 15 20 47](https://github.com/user-attachments/assets/636cdcb7-b0b4-4ebc-b24f-e29ceb3f55eb)

---

### Notities

- Rechthoek kleurvlak
- Aangevuld met naam en hex waarde.
- In Figma staat er soms een tekst in het kleurvlak.

**Comment by jeffreylauwers** [comment]
## Den Haag

Niet gevonden als component. Maar er worden wel voorbeelden van kleuren getoond voor het design system.

- [Storybook](https://nl-design-system.github.io/denhaag/?path=/docs/den-haag-colors--docs)
- [Figma huidig](https://www.figma.com/design/JpoY3waVoQGlLQzQXTL9nn/%F0%9F%8C%BB-Den-Haag-Design-System?node-id=3497-16154&t=BIQjKgSgX61hHBoZ-4)
- [Figma WIP](https://www.figma.com/design/x4RkF6BIdrNZbh7D53NTzB/%F0%9F%92%A0-NLDS---Den-Haag---Bibliotheek?node-id=3257-703&t=NxfzezHLAy5YSC7W-4)

---

### Screenshots / Varianten

**Storybook**

Color palette
(Screenshot bevat niet alle kleuren)
![Scherm­afbeelding 2024-09-06 om 15 23 36](https://github.com/user-attachments/assets/f1f80845-a588-4d1d-9170-e2ed3b7d52d4)


---

### Notities

- In Storybook vult het kleurvlak zich uit binnen een balk.
- Aangevuld met naam en hsl waarde.

**Comment by jeffreylauwers** [comment]
## DSO

Niet gevonden als component. Maar er worden wel voorbeelden van kleuren getoond voor het design system.

[Documentatie website](https://www.dso-toolkit.nl/62.8.4/fundament/kleuren)

---

### Screenshots / Varianten

**Documentatie website**
(Screenshot bevat niet alle kleuren)
![Scherm­afbeelding 2024-09-06 om 15 30 32](https://github.com/user-attachments/assets/d961bc38-7adb-4ea5-b5ab-cda96139b51f)


---

### Notities

- Vierkant kleurvlak.
- Aangevuld met naam, rgb en hex waarde. Informatie staat in het kleurvlak.

**Reply by julezrulez** [reply]
In dit voorbeeld zie je ten opzichte van de meeste andere voorbeelden meteen welke tekstkleur je moet gebruiken om aan de contrastregel uit WCAG te voldoen.

**Comment by jeffreylauwers** [comment]
## DUO

Niet gevonden als component. Maar er worden wel voorbeelden van kleuren getoond voor het design system.

[Documentatie website](https://uno.dfront.rijkscloud.nl/#/richtlijnen/colors)

---

### Screenshots / Varianten

**Documentatie website**
(Screenshot bevat niet alle kleuren)
![Scherm­afbeelding 2024-09-09 om 07 10 06](https://github.com/user-attachments/assets/7156c519-dfe6-4496-930a-fd7b5bcd4370)

---

### Notities

- Rechthoeking kleurvlak
- Aangevuld met naam, hex waarde. Informatie staat in het kleurvlak.

**Reply by julezrulez** [reply]
In dit voorbeeld zie je ten opzichte van de meeste andere voorbeelden meteen welke tekstkleur je moet gebruiken om aan de contrastregel uit WCAG te voldoen.

**Comment by jeffreylauwers** [comment]
## OpenGemeenten

Niet gevonden als component. Maar er worden wel voorbeelden van kleuren getoond voor het design system.

[Patternlab](https://patternlab.lenteveld.nl/index.html)

---

### Screenshots / Varianten

**Patternlab**
(Screenshot bevat niet alle kleuren)

![Scherm­afbeelding 2024-09-09 om 07 13 57](https://github.com/user-attachments/assets/213fdb4b-5381-435b-8c6a-19905c018473)


---

### Notities

- Rechthoekig kleurvlak met afgeronde hoeken.
- Omkaderd en aangevuld met naam.

**Comment by jeffreylauwers** [comment]
## Rotterdam

Niet gevonden als component. Maar er worden wel voorbeelden van kleuren getoond voor het design system.

- [Storybook](https://nl-design-system.github.io/rotterdam/?path=/docs/rotterdam-color--docs)
- [Figma](https://www.figma.com/design/ZWSC4gCrOXRUR9UX3aoZ8x/Gemeente-Rotterdam-Design-System?node-id=415-12199&t=sxUAr7B2WLiiRwC8-4)

---

### Screenshots / Varianten

**Storybook**
(Screenshot bevat niet alle kleuren)
![Scherm­afbeelding 2024-09-09 om 07 16 36](https://github.com/user-attachments/assets/f4ef6702-737d-4809-9ecb-4f0996a4cde9)

Figma
(Screenshot bevat niet alle kleuren)
![Scherm­afbeelding 2024-09-09 om 07 24 48](https://github.com/user-attachments/assets/58609423-3681-404c-b3b8-acd30f9707af)

---

### Notities

- In Storybook vult het kleurvlak zich uit binnen een balk.
- In Figma is kleurvlak een vierkant met afgeronde hoeken.
- Aangevuld met naam, hex waarde.

**Comment by jeffreylauwers** [comment]
## RVO

Niet gevonden als component. Maar er worden wel voorbeelden van kleuren getoond voor het design system.

- [Documentatie website](https://nl-design-system.github.io/rvo/docs/design-tokens/color/)
- [Storybook](https://nl-design-system.github.io/rvo/?path=/docs/huisstijl-color--docs)

---

### Screenshots / Varianten

**Storybook**
(Screenshot bevat niet alle kleuren)

![Scherm­afbeelding 2024-09-09 om 09 51 57](https://github.com/user-attachments/assets/f38f98cb-cace-485a-9bb6-16661563eb91)


---

### Notities

- Kleurvlak als cirkel
- Aangevuld met naam, hex en rgb waarde, aangevuld met CSS variabele.

**Comment by jeffreylauwers** [comment]
## Utrecht

- [Documentatie website](https://nl-design-system.github.io/utrecht/docs/tmp/huisstijl/kleuren/)
- [Storybook](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-color-sample--docs)
- [GitHub](https://github.com/nl-design-system/utrecht/tree/main/components/color-sample)

---

### Screenshots / Varianten

**Documentatie website**
(Screenshot bevat niet alle kleuren)
![Scherm­afbeelding 2024-09-09 om 09 54 57](https://github.com/user-attachments/assets/8f0e3d58-1bab-4b92-80ed-a8a859b8258d)

**Storybook**

Default
![Scherm­afbeelding 2024-09-09 om 09 55 47](https://github.com/user-attachments/assets/1f3f137c-e092-4fcc-8f3d-cc5c6be1231a)

---

### Notities

- Documentatie beschikbaar
- Naast gebruik zoals bij andere organisaties wordt bij Urecht ook de toepassing voor een Legenda en combinatie met een color input component genoemd.

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Color sample op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=color+sample)

**Comment by Rozerinay** [comment]
## ✨ Dit component is nu Community ✨

Help je mee hem door de Candidate stappen te krijgen?
[Color Sample op het Candidate bord](https://github.com/orgs/nl-design-system/projects/32/views/1?filterQuery=color+sample)

**Comment by rianrietveld** [comment]
@jeffreylauwers at al
We (Marjon en ik) hebben onderzoek gedaan met screen readers naar de constructie

```
<div class="voorbeeld-theme">
    <svg 
        aria-labelledby=":r0:" 
        role="img" 
        xmlns="http://www.w3.org/2000/svg" height="16" width="16" class="nl-color-sample" style="color: rgb(255, 20, 147);">
        <path d="M0 0H32V32H0Z" fill="currentcolor"></path>
    </svg>
        
    <p class="nl-paragraph" id=":r0:">De kleur "#ff1493".</p>
</div>
```


Zie: https://candidate-storybook-test-git-refactor-c-118d12-nl-design-system.vercel.app/?path=/docs/componenten-color-sample--documentatie.

Resultaat:
- VoiceOver in Safari leest het goed voor, de SVG en de paragraaf.
- NVDA in Edge leest alleen de kleur voor bij de SVG en slaat de labelledby tekst over, dus dat is eigenlijk perfect.
- JAWS leest het goed voor, de SVG en de paragraaf, net zoals VoiceOver.

Twee keer voorlezen is eigenlijk geen probleem, je ziet ook twee keer een aanwijzing voor de kluer.
Deze constructie wordt dus door screen readers voldoende ondersteund.

Voorwaarde voor deze constructie:

- De `p` voor de `labelledby` moet altijd gevuld zijn met de beschrijving van de kleur, anders heeft de SVG geen toegankelijke naam.
- De SVG moet het attribuut `img` hebben.

**Reply by MarjonBakker** [reply]
> @jeffreylauwers at al We (Marjon en ik) hebben onderzoek gedaan met screen readers naar de constructie

> Resultaat:
> 
>     * NVDA in Edge leest alleen de kleur voor bij de SVG en slaat de labelledby tekst over, dus dat is eigenlijk perfect.

@rianrietveld Ik kan het niet reproduceren dat NVDA iets overslaat.

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Candidate ✨

Help je mee hem door de Hall of Fame stappen te krijgen?
[Color Sample op het Hall of Fame bord](https://github.com/orgs/nl-design-system/projects/30/views/1?filterQuery=color+sample)

**Comment by eelcobosklopper** [comment]
Het kernteam heeft besloten om de Color Sample Retroactief door de voorbereidingsfase van de candidate stap te halen. [Zie dit issue](https://github.com/nl-design-system/candidate/issues/1288). Deze discussion gaan we dus weer wat aanzwengelen om feedback uit de community op te halen. Doe vooral (weer) mee!

**Comment by eelcobosklopper** [comment]
## 1. Wie gebruikt nu de candidate color sample?
Gebruiken je de candidate color sample al en waarvoor precies? In storybook of een andere applicatie/site?

**Reply by bartveneman** [reply]
Expertteam Digitale Toegankelijkheid gebruikt de color sample veelvuldig in de hele Theme Wizard applicatie (https://theme-wizard.nl-design-system-community.nl/). We hebben de candidate in een `<clippy-color-sample>` gestopt, in storybook gezet en gebruiken we in de hele app op tientallen plekken.

- usage: https://github.com/search?q=repo%3Anl-design-system%2Ftheme-wizard+clippy-color-sample&type=code
- storybook: https://clippy-storybook-nl-design-system.vercel.app/?path=/docs/clippy-color-sample--documentatie

**Reply by bramsmulders** [reply]
In de `clippy-color-sample` hebben we [deze bug](https://github.com/nl-design-system/candidate/issues/1286) ook gefixed. Lijkt tot nu toe goed te werken. 👍

**Comment by eelcobosklopper** [comment]
## 2. Reageer in de thread! Zijn er nog features die jullie missen?
Denk aan het toevoegen van content aan het element, dat het in print nog goed toont etc.

**Comment by eelcobosklopper** [comment]
## 3. Reageer in de thread: Wat vinden jullie van de checkerboard achtergrond? Heeft dit nut? 

De huidige implementatie heeft een checkerboard achtergrond. Hierdoor kan transparantie van de kleur beter duidelijk gemaakt worden.
<img width="425" height="131" alt="Image" src="https://github.com/user-attachments/assets/cd5791cf-57a2-4444-9bd2-482e748d4fb2" />

Worden transparante kleuren gebruikt?

**Reply by bartveneman** [reply]
+1 voor deze achtergrond. In de context van de Theme Wizard helpt dit om ontoegankelijke keuzes te vermijden die door semitransparante kleuren veroorzaakt kunnen worden. In de Theme Wizard kunnen we zelf wel detecteren dat een kleur transparant is, maar deze background helpt ook om dat aan gebruikers duidelijk te maken.

**Comment by eelcobosklopper** [comment]
## 4. Reageer in de thread: Meerwaarde van zowel svg als andere content kunnen renderen

Zou je de optie willen hebben om naast een SVG ook content op een gekleurde achtergrond te willen tonen? Bijvoorbeeld:

<img width="118" height="107" alt="Image" src="https://github.com/user-attachments/assets/e68660dd-f627-4a27-b2be-46c29543af84" />
